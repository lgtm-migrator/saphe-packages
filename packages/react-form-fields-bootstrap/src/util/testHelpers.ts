import useForm, { Field } from '@saphe/react-form';
import { renderHook } from '@testing-library/react-hooks';
import { ReactElement } from 'react';
import { BootstrapFieldPack } from '..';

interface FormState {
  Form: () => ReactElement;
}

export const renderForm = (hook: () => FormState): (() => ReactElement) =>
  renderHook(hook).result.current.Form;

export const minimalTextFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalTextFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      textField: {
        type: Field.TEXT,
        label: 'Text Field',
        description: props?.description,
        // validation: {
        //   required:
        // }
      },
    },
  });

export const minimalTextAreaFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalTextAreaFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      textAreaField: {
        type: Field.TEXT_AREA,
        label: 'Textarea Field',
        description: props?.description,
      },
    },
  });

export const minimalSelectFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalSelectFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      selectField: {
        type: Field.SELECT,
        label: 'Select Field',
        description: props?.description,
        options: [
          {
            label: 'Option 1',
            value: 'option-1',
          },
          {
            label: 'Option 2',
            value: 'option-2',
          },
        ],
      },
    },
  });

export const minimalCheckBoxFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalCheckBoxFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      checkBoxField: {
        type: Field.CHECK,
        label: 'Checkbox Field',
        description: props?.description,
      },
    },
  });

export const minimalNumberFieldForm = (props?: {
  description?: string;
}): FormState =>
  useForm({
    name: 'minimalNumberFieldForm',
    fieldPack: BootstrapFieldPack,
    fields: {
      numberField: {
        type: Field.NUMBER,
        label: 'Number Field',
        description: props?.description,
      },
    },
  });
