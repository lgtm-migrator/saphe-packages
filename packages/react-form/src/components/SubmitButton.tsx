import React, { ReactElement } from 'react';
import { SubmitButtonProps } from '../utils/fieldPropTypes';
import { FieldPack } from '../utils/helperTypes';

interface Props extends SubmitButtonProps {
  fieldPack?: FieldPack;
}

export default function SubmitButton(props: Props): ReactElement {
  return props.fieldPack?.SubmitButton ? (
    <props.fieldPack.SubmitButton isSubmitting={props.isSubmitting} />
  ) : (
    <button type="submit" disabled={props.isSubmitting}>
      {props.isSubmitting ? 'Submitting...' : 'Submit'}
    </button>
  );
}
