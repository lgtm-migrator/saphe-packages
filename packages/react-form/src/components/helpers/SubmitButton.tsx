import React, { ReactElement } from 'react';
import { AddFieldPack } from '../../utils/helperTypes';
import { SubmitButtonProps } from '../../utils/propTypes';

export default function SubmitButton(
  props: AddFieldPack<SubmitButtonProps>,
): ReactElement {
  return props.fieldPack?.SubmitButton ? (
    <props.fieldPack.SubmitButton isSubmitting={props.isSubmitting} />
  ) : (
    <button type="submit" disabled={props.isSubmitting}>
      {props.isSubmitting ? 'Submitting...' : 'Submit'}
    </button>
  );
}
