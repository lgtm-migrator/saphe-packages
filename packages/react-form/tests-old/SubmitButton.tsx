import SubmitButton from '../src/components/helpers/SubmitButton';
import { matchSnapshot } from './testHelpers';

describe('Submit Button', () => {
  it('renders', () => {
    matchSnapshot(<SubmitButton type="button" isLoading={false} disabled={false} label="Submit" />);
    matchSnapshot(
      <SubmitButton type="submit" isLoading={true} disabled={false} label="Processing..." />,
    );
  });
});
