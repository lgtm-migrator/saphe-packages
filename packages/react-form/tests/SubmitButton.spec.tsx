import React from 'react';
import SubmitButton from '../src/components/helpers/SubmitButton';
import { matchSnapshot } from './testHelpers';

describe('Submit Button', () => {
  it('renders', () => {
    matchSnapshot(<SubmitButton isSubmitting={false} />);
    matchSnapshot(<SubmitButton isSubmitting={true} />);
  });
});
