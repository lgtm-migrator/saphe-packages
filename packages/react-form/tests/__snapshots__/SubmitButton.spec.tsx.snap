// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit Button renders 1`] = `
<button
  disabled={false}
  type="button"
>
  Submit
</button>
`;

exports[`Submit Button renders 2`] = `
<button
  disabled={true}
  type="submit"
>
  Processing...
</button>
`;
